### PR TITLE
Implement etcd backup & restore of OSS on alicloud

### PR DESCRIPTION
**Pull request by minchaow**
**What this PR does / why we need it**:
This PR is to implement etcd backup & restore of OSS on alicloud.
**Which issue(s) this PR fixes**:
Fixes #
None
**Special notes for your reviewer**:
None

### CODE REVIEW

File: charts/seed-terraformer/charts/alicloud-backup/templates/_main.tf
Comment (rfranzke) [inline charts/seed-terraformer/charts/alicloud-backup/templates/_main.tf:33]: Adding an `output` field won't make `terraform plan` identify a difference.

Can you please add a similar section like https://github.com/gardener/gardener/blob/master/charts/seed-terraformer/charts/openstack-infra/templates/_main.tf#L88-L96 to trigger a `terraform apply` run?

```
// We have introduced new output variables. However, they are not applied for
// existing clusters as Terraform won't detect a diff when we run `terraform plan`.
// Workaround: Providing a null-resource for letting Terraform think that there are
// differences, enabling the Gardener to start an actual `terraform apply` job.
resource "null_resource" "outputs" {
  triggers = {
    recompute = "outputs"
  }
}
```

File: pkg/operation/cloudbotanist/alicloudbotanist/controlplane.go
Comment (rfranzke) [inline pkg/operation/cloudbotanist/alicloudbotanist/controlplane.go]:
```suggestion
		"schedule":         b.ShootBackup.Schedule,
```

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: pkg/operation/cloudbotanist/alicloudbotanist/controlplane.go
Comment (minchaow) [reply pkg/operation/cloudbotanist/alicloudbotanist/controlplane.go]: Addressed the comment. However, noticed in `controlplane.go` of other type cloud botanists, `Operation` is also called first and then `ShootBackup` is called from it.

**Review by rfranzke — APPROVED** [review]
/lgtm

File: charts/seed-terraformer/charts/alicloud-backup/templates/_main.tf
Comment (swapnilgm) [inline charts/seed-terraformer/charts/alicloud-backup/templates/_main.tf:14]: Please fill up the following fields:
```
location: " {{ .Values.alicloud.region }}"
storage_class: "Standard"
```

**Review by swapnilgm — CHANGES_REQUESTED** [review]
Config generation part LGTM. I have suggested some changes to terraform please address them. 
Apart from that, please disable destroying bucket [here](https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/alicloudbotanist/infrastructure.go#L82) for now since [alicloud OSS doesn't support deletion on non-empty buckets](https://www.alibabacloud.com/help/doc-detail/31846.htm?spm=a2c63.l28256.b99.39.73c05139yxJeQf) via terraform. With terraform such case is handled by passing `force_destroy=true` flag in bucket resource with other cloud provider. Please raise PR on [terraform-provider-alicloud](https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/resource_alicloud_oss_bucket.go#L724) where; if bucket is not empty and `force_destroy` flag is enabled then change bucket deletion logic to delete all object in bucket and then try deleting bucket.

We had followed similar approach with openstack before. WDYS @rfranzke ?

**Comment by rfranzke** [comment]
Thank you @swapnilgm!
So, we cannot really merge it right now - as Alicloud Terraform provider does not yet support (non-empty) deletion of buckets then we would end up with lots of "orphaned" OSS buckets for deleted shoots, right?
So, first @minchaow has to implement the force-deletion logic in Terraform before this code can be merged, right?

FYI: You don't need to use Terraform for creation/deletion of buckets. You could also use the Alicloud Go SDK to do that, but then you have to handle all the features Terraform provides (retry logic, dependency graphs, state management, etc.) on your own (although, I guess for bucket creation/deletion it's not that much).

@minchaow please tell us how you are going to proceed. Thank you.

**Comment by swapnilgm** [comment]
> So, we cannot really merge it right now - as Alicloud Terraform provider does not yet support (non-empty) deletion of buckets then we would end up with lots of "orphaned" OSS buckets for deleted shoots, right?

Yup. I just said yes to that approach cause we were ok with manual deletion or orphaned Buckets in opnestacks for time being.

**Comment by rfranzke** [comment]
Yes, but that was a different situation. If @minchaow is also fine with manual deletion then let's merge it, but I would assume it's too much manual work.

**Comment by swapnilgm** [comment]
Agreed.

**Comment by swapnilgm** [comment]
If its not urgent, then i would also prefer to wait for adding support in terraform-provider-alicloud

**Comment by rfranzke** [comment]
👍 so, @minchaow please tell us how you want to proceed.

**Comment by minchaow** [comment]
Hi @rfranzke, @swapnilgm, sorry for my late reply and thanks for good suggestions from your side. I agree, it is better to merge this change when "force-destroy" support in terraform-provider-alicloud is ready. We already have some contacts with Alicloud, and will discuss with them about this feature asap.

@jia-jerry , what's your comments?

**Comment by jia-jerry** [comment]
Agree.

**Comment by rfranzke** [comment]
Alright. Then let's close the PR for now. Please re-open as soon as force-destroy is working in the terraform-provider-alicloud.

**Pull request closed without merging**
